connectors/sources/sharepoint_server.py

```diff
@@ -515,8 +657,246 @@ def get_default_configuration(cls):
+                "display": "toggle",
+                "label": "Fetch unique list permissions",
+                "order": 10,
+                "tooltip": "Enable this option to fetch unique list permissions. This setting can increase sync time. If this setting is disabled a list will inherit permissions from its parent site.",
+                "type": "bool",
+                "value": True,
+            },
+            "fetch_unique_list_item_permissions": {
+                "depends_on": [{"field": "use_document_level_security", "value": True}],
+                "display": "toggle",
+                "label": "Fetch unique list item permissions",
+                "order": 11,
+                "tooltip": "Enable this option to fetch unique list item permissions. This setting can increase sync time. If this setting is disabled a list item will inherit permissions from its parent site.",
+                "type": "bool",
+                "value": True,
+            },
         }
 
+    def _dls_enabled(self):
+        if (
+            self._features is None
+            or not self._features.document_level_security_enabled()
+        ):
+            return False
+
+        return self.configuration["use_document_level_security"]
+
+    def _decorate_with_access_control(self, document, access_control):
+        if self._dls_enabled():
+            document[ACCESS_CONTROL] = list(
+                set(document.get(ACCESS_CONTROL, []) + access_control)
+            )
+
+        return document
+
+    def access_control_query(self, access_control):
+        return es_access_control_query(access_control)
+
+    async def _user_access_control_doc(self, user):
+        login_name = user.get("LoginName")
+        prefixed_mail = _prefix_email(user.get("Email"))
+        prefixed_title = _prefix_user(user.get("Title"))
+        prefixed_user_id = _prefix_user_id(user.get("Id"))
+        prefixed_login_name = _prefix_login_name(login_name)
+
+        access_control = [
+            prefixed_mail,
+            prefixed_title,
+            prefixed_user_id,
+            prefixed_login_name,
+        ]
+
+        return {
+            "_id": login_name,
+            "identity": {
+                "email": prefixed_mail,
+                "title": prefixed_title,
+                "user_id": prefixed_user_id,
+                "login_name": prefixed_login_name,
+            },
+            "created_at": iso_utc(),
+        } | self.access_control_query(access_control)
+
+    async def _access_control_for_member(self, member):
+        principal_type = member.get("PrincipalType")
+        is_group = principal_type != 1 if principal_type else False
+        user_access_control = []
+
+        if is_group:
+            group_name = member.get("Title")
+            self._logger.debug(f"Detected group '{group_name}'.")
+            group_id = _get_login_name(member.get("LoginName"))
```

Comment:
I don't know if this debug log adds much. I think it could be more valuable if it included more information, and was combined with another debug log within the `else` statement. Otherwise we're only logging group names for some reason. 

```suggestion
            group_name = member.get("Title")
            group_id = _get_login_name(member.get("LoginName"))
            self._logger.debug(f"Detected member '{group_id}' with name '{group_name}' and principal type '{principal_type}', processing as group.")
```

See my other comment below for a similar debug log to complement this.